lib/Backend/arm64/LowerMD.cpp

```diff
@@ -8372,6 +8316,40 @@ void LowererMD::GenerateFloatTest(IR::RegOpnd * opndSrc, IR::Instr * insertInstr
     insertInstr->InsertBefore(instr);
 }
 
+IR::RegOpnd* LowererMD::CheckFloatAndUntag(IR::RegOpnd * opndSrc, IR::Instr * insertInstr, IR::LabelInstr* labelHelper)
+{
+    IR::Opnd* floatTag = IR::AddrOpnd::New((Js::Var)Js::FloatTag_Value, IR::AddrOpndKindConstantVar, this->m_func, /* dontEncode = */ true);
+    IR::RegOpnd* regOpndFloatTag = IR::RegOpnd::New(TyUint64, this->m_func);
```

Comment:
Why is this an AddrOpnd and not an IntConstOpnd?

Reply:
Note that FloatTag_Value is directly encodeable as an immediate to logical operations (AND/OR/EOR), so might be most optimal to leave it as an immediate in the operations below.

Reply:
It believe the reason we use AddrOpnd is to help with IR dumping. An addr that matches that is uniquely FloatTag, where as a number that matches that might not be.